forui/lib/src/widgets/text_field/text_field.dart

```diff
@@ -983,6 +920,194 @@ class FTextField extends StatefulWidget {
          'To fix, set the initial text directly in the controller.',
        );
 
+  // Default toggle builder for password visibility
+  static Widget _defaultToggleBuilder(BuildContext context, FTextFieldStyle style, Set<WidgetState> states) {
+    // This should not be called directly since it requires a controller
+    throw StateError(
+      'Default toggle builder should not be called without a visibility controller. '
+      'This is an internal implementation error.',
+    );
+  }
+
+  /// Creates a [FTextField] configured for passwords with a visibility toggle.
+  ///
+  /// This creates a password field with an eye icon that allows users to toggle
+  /// between showing and hiding the password text. The toggle is enabled by default
+  /// but can be disabled by setting [toggleBuilder] to null.
+  ///
+  /// [visibilityController] controls the password visibility state. If not provided,
+  /// an internal controller will be created and managed automatically. When using
+  /// a custom [toggleBuilder], you must provide your own [visibilityController]
+  /// since the custom builder needs to manage the visibility state.
+  ///
+  /// [toggleBuilder] builds the toggle button for password visibility. If null,
+  /// no toggle will be shown and the password will remain obscured. If not provided
+  /// (defaults to not null), a default eye icon toggle will be shown.
+  ///
+  /// [autofillHints] defaults to [AutofillHints.password]. It should be overridden
+  /// with [AutofillHints.newPassword] when handling the creation of new passwords.
+  ///
+  /// Example:
+  /// ```dart
+  /// // Basic usage with automatic toggle
+  /// FTextField.password()
+  ///
+  /// // With custom controller (required for custom toggle)
+  /// final controller = ValueNotifier<bool>(false);
+  /// FTextField.password(
+  ///   visibilityController: controller,
+  ///   toggleBuilder: (context, style, states) => IconButton(
+  ///     onPressed: () => controller.value = !controller.value,
+  ///     icon: Icon(controller.value ? Icons.visibility_off : Icons.visibility),
+  ///   ),
+  /// )
+  ///
+  /// // Disable toggle completely
+  /// FTextField.password(
+  ///   toggleBuilder: null,
+  /// )
+  /// ```
+  static Widget password({
```

Comment:
I think we can create a private subclass of FTextField. That way we can use a redirecting factory constructor thereby solving this issue.